src/dataset_tools/bq_squashfs/create_webdataset.py

```
@@ -0,0 +1,537 @@
+    """Pack one split's NVMe shard dirs → Lustre tar files. Returns total bytes."""
+    split_shards = nvme_shards / split
+    non_empty = [
+        split_shards / f"shard_{s:06d}"
+        for s in range(n_shards)
+        if (split_shards / f"shard_{s:06d}").exists()
+        and any((split_shards / f"shard_{s:06d}").iterdir())
+    ]
+    log(f"  [{split}] packing {len(non_empty):,} non-empty shards  "
+        f"(mode='{tar_mode}', {pack_workers} workers)")
+
+    rng = random.Random(seed)
```

Ruff (0.15.15)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)